recirq/otoc/loschmidt/tilted_square_lattice/analysis.py

```diff
@@ -0,0 +1,194 @@
+                         for result in results.executable_results])
+
+
+def loschmidt_results_to_dataframe(results: ExecutableGroupResult) -> pd.DataFrame:
+    """Process an `ExecutableGroupResult`.
+
+    This function performs the data analysis using `to_ground_state_prob` and
+    extracts the most relevant quantities for a dataframe suitable for
+    grouping, selecting, and plotting.
+    """
+
+    def _to_record(result: ExecutableResult,
+                   rt_config: QuantumRuntimeConfiguration,
+                   shared_rt_info: SharedRuntimeInfo) -> Dict:
+        success_prob = to_ground_state_prob(result.raw_data)
+        spec = cast(TiltedSquareLatticeLoschmidtSpec, result.spec)
+
+        return {
+            'run_id': shared_rt_info.run_id,
+            'width': spec.topology.width,
+            'height': spec.topology.height,
+            'n_qubits': spec.topology.n_nodes,
+            'macrocycle_depth': spec.macrocycle_depth,
+            'q_area': (spec.macrocycle_depth * 4 * 2) * spec.topology.n_nodes,
+            'instance_i': spec.instance_i,
+            'n_repetitions': spec.n_repetitions,
+            'success_probability': success_prob,
+            'processor_id': rt_config.processor_record.processor_id,
+        }
+
+    return _results_to_dataframe(results, _to_record)
+
+
+def fit_vs_macrocycle_depth(df):
+    """Perform a fitting vs macrocycle depth given a loschmidt results dataframe.
+
+    Args:
+        df: The dataframe from `loschmidt_results_to_dataframe`.
+
+    Returns:
+        fitted_df: A new dataframe containing fit parameters.
+        exp_ansatz_vs_macrocycle_depth: The function used for the fit.
+    """
+
+    def exp_ansatz_vs_macrocycle_depth(macrocycle_depths, a, f):
+        return a * np.exp(np.log(f) * 4 * 2 * macrocycle_depths)
+
+    def _fit(row):
+        (a, f), pcov = curve_fit(
+            exp_ansatz_vs_macrocycle_depth,
+            xdata=row['macrocycle_depth'],
+            ydata=row['success_probability'],
+        )
+        a_err, f_err = np.sqrt(np.diag(pcov))
+        row['a'] = a
+        row['f'] = f
+        row['a_err'] = a_err
+        row['f_err'] = f_err
+        return row
+
+    y_cols = ['instance_i', 'macrocycle_depth', 'q_area', 'success_probability']
+    agged, _ = groupby_all_except(
+        df,
+        y_cols=y_cols,
+        agg_func=list,
+    )
+    fitted_df = agged.apply(_fit, axis=1) \
+        .drop(y_cols, axis=1)
+    return fitted_df, exp_ansatz_vs_macrocycle_depth
+
+
+def fit_vs_q_area(df):
+    """Perform a fitting vs "quantum area", i.e. depth*n_qubits.
+
+    Args:
+        df: The dataframe from `loschmidt_results_to_dataframe`.
+
+    Returns:
+        fitted_df: A new dataframe containing fit parameters.
+        exp_ansatz_vs_qarea: The function used for the fit.
+    """
+
+    def exp_ansatz_vs_q_area(q_area, a, f):
+        return a * np.exp(np.log(f) * q_area)
+
+    def _fit(row):
```

Comment:
This function can be shared

Reply:
not really. the xdata column names are different